### PR TITLE
Branch edit book command

### PR DESCRIPTION
**Pull request by yuqitanyq**
Create Edit Book Command functionality with sufficient testing to make sure the feature works well.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/book/EditBookCommand.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(bookName, isbn, authors, tags);
+        }
+
+        public void setBookName(BookName bookname) {
+            this.bookName = bookname;
+        }
+
+        public Optional<BookName> getBookName() {
+            return Optional.ofNullable(bookName);
```
Comment (bingsen0806) [inline src/main/java/seedu/address/logic/commands/book/EditBookCommand.java:155]: Good use of Optional

File: src/main/java/seedu/address/logic/parser/book/EditBookCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+        EditBookDescriptor editBookDescriptor = new EditBookDescriptor();
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            editBookDescriptor.setBookName(ParserUtil.parseBookName(argMultimap.getValue(PREFIX_NAME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_ISBN).isPresent()) {
+            editBookDescriptor.setIsbn(ParserUtil.parseIsbn(argMultimap.getValue(PREFIX_ISBN).get()));
+        }
+
+        parseAuthorsForEdit(argMultimap.getAllValues(PREFIX_AUTHOR)).ifPresent(editBookDescriptor::setAuthors);
```
Comment (bingsen0806) [inline src/main/java/seedu/address/logic/parser/book/EditBookCommandParser.java:58]: Good application of SLAP principles and use high level code

File: src/test/java/seedu/address/model/book/EditBookCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+ * Contains integration tests (interaction with the Model) and unit tests for EditBookCommand.
+ */
+public class EditBookCommandTest {
+
+    private Model model;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(TypicalAddressBook.getTypicalAddressBook(), new UserPrefs());
+    }
```
Comment (bingsen0806) [inline src/test/java/seedu/address/model/book/EditBookCommandTest.java:44]: Good use of @BeforeEach to ensure that the model is reset after every test

File: src/test/java/seedu/address/testutil/EditBookDescriptorBuilder.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        bookDescriptor.setAuthors(SampleDataUtil.getAuthorList(authors));
+        return this;
+    }
+
+    /**
+     * Parses the {@code tags} into a {@code Set<Tag>} and set it to the {@code EditBookDescriptor}
+     * that we are building.
+     */
+    public EditBookDescriptorBuilder withTags(String... tags) {
+        Set<Tag> tagSet = Stream.of(tags).map(Tag::new).collect(Collectors.toSet());
```
Comment (bingsen0806) [inline src/test/java/seedu/address/testutil/EditBookDescriptorBuilder.java:68]: Good use of Java streams!

**Review by bingsen0806 — APPROVED** [review]
LGTM! Good testing overall and good use of SLAP princples and Java streams!

File: src/main/java/seedu/address/logic/commands/book/EditBookCommand.java [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+                return false;
+            }
+
+            // state check
+            EditBookDescriptor e = (EditBookDescriptor) other;
+
+            return getBookName().equals(e.getBookName())
+                    && getIsbn().equals(e.getIsbn())
+                    && getAuthors().equals(e.getAuthors())
+                    && getTags().equals(e.getTags());
```
Comment (yl-ang) [inline src/main/java/seedu/address/logic/commands/book/EditBookCommand.java:209]: Good implementation of the equals method and good use of guard clauses to ensure that the reader is not distracted from the alternative paths.

**Review by yl-ang — APPROVED** [review]
LGTM!
Good use of guard clauses, methods are not long (around 30 or lesser lines) and code quality is up to standard. No observable errors or typos in the java docs.

Keep up the good work 💯

**Pull request merged**